SendLog.py

```diff
@@ -126,90 +220,576 @@ def event_stream(xcos_file_id):
+                print("<data column=\"0\" line=\"4\" value=\"2\"/>")
+                print("</ScilabString>")
+                print("<ScilabDouble as=\"realParameters\" height=\"0\" width=\"0\"/>")
+                print("<ScilabDouble as=\"integerParameters\" height=\"105\" width=\"1\">")
+                tk_count = tk_count + 1
+                # The remaining part of the block is read from the Read_Content.txt file and written to the xml file
+                read_file = open("Read_Content.txt", "r")
+                for line_content in read_file:
+                    print(line_content,end= '')
+            print(line,end = '')
+
+
+        # Changing the file extension from xml to xcos
+        base_filename = os.path.splitext(temp_file_xml_name)[0]
+        os.rename(temp_file_xml_name, base_filename + ".xcos")
+        source_folder = os.getcwd()
+        destination_folder = source_folder + "/uploads/"
+
+
+        # Move the xcos file to uploads directory
+        folder_file_content = filter(os.path.isfile, os.listdir( os.curdir ) )
+        for file in folder_file_content:
+            if file.endswith(".xcos"):
+                shutil.copy(file, destination_folder)
+                subprocess.Popen(["rm",file])# After moving to the required folder deleting the xcos file.
+        temp_file_xml_name = base_filename + ".xcos"
+        xcos_file_list.append(temp_file_xml_name)
+        workspace_list.append(workspace_counter)
+        return str(client_id)
+    else:
+        return "error"
+
+@app.route('/UpdateTKfile', methods=['POST'])    
+def UpdateTKfile():		# '=> Dattatreya <='  function which makes the initialazation and updation of the files with obtained new value
 	# Get the file
 	file = request.files['file']
+
 	# Check if the file is not null
 	if file:
-		# Make the filename safe, remove unsupported chars
-		client_id = len(xcos_file_list)
-		filename = str(client_id)+".xcos"
-		# Move the file form the temporal folder to
-		# the upload folder we setup
-		file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
-		xcos_file_list.append(filename)
-		return str(client_id)
+		filename1 = Details.uid+'_val.txt'
+		file.save(os.path.join(app.config['VALUES_FOLDER'], filename1)) # saves the file in values folder
+		n = findFile()
+		if n==0:
+		   getDetailsThread() # starts the thread
+
+		elif n==1:
+		   stopDetailsThread() # stops the thread
+		
+		return ""
 	else:
 		return "error"
+@app.route('/importXcos')   # '=> Dattatreya <=' function to take the request data when user click the url link (view example)
+def importXcos():
+
+	
+	Details.names[Details.uid+"eid"] = request.args.get('eid')	# get the eid from the url
+	
+	with open("webapp/xcos/"+str(Details.names[Details.uid+"eid"])+".xcos") as inf:
+		Details.names[Details.uid+"xcosContents"] = inf.read()	# read the xcos file contents
+	import bs4
+	with open("webapp/index.html") as inf:
```

Comment:
Why are you changing your web templates dynamically? You need to use some sort of templating language.